File: Examples/Moving Ball/MovingBall.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Feb 22 11:23:28 2021

@author: MattPetersonsAccount
"""
from psychopy import visual, core, event
import sys

# make a new floating window that is filled with black
win = visual.Window( fullscr = False, color=(-1, -1, -1), units="height")
#=============================
def main():
	myCirc = visual.Circle(win,
             	units="height",
				radius = 0.1,               # 20% of the screen height
				edges = 32,
                	pos=( -1, 1),	# screen center
                	fillColor = [+1,-1,-1],	# r,g,b
                	lineColor = "yellow",       	# has color names
                	interpolate=True)
	myCirc.draw()                           # everything draws into memory
	win.flip()                              # copies to screen
	core.wait(2)                            # wait one second


	for i in range(10):
		x = -.5 + (i*0.1)
		myCirc.pos = (x, 0)
		myCirc.draw()                           # everything draws into memory
		#print(x)
		win.flip()                              # copies to screen
		core.wait(0.1)


	win.close()
	core.quit()
	sys.exit(0)
	#exit()

#===============================================================



if __name__ == "__main__":
	main()

File: Examples/Moving Ball/MovingBall.jl
using PsychExpAPIs
using SimpleDirectMediaLayer
using SimpleDirectMediaLayer.LibSDL2
# Moving Ball Exampile



#-=============================
function main()
	InitPsychoJL()
	# make a new floating window using the PsychoPy coordinate space and and color space
	win = Window( [1280, 720], false; colorSpace = "PsychoPy", coordinateSpace = "PsychoPy", timeScale = "seconds")			#	2560, 1440			[1000,1000]

	myCirc = Circle(win,
					[ 0.0, 0.0],					        # screen center
                	0.1,					                # radius is 10% of the screen height
					fillColor = [+1.0,-1.0,-1.0, +1.0],     # r,g,b, alpha
                	lineColor = "yellow",					# has color names
					fill = true)
  
	draw(myCirc)                           # everything draws into memory
	flip(win)                              # copies to screen
	waitTime(win, 0.5)                            # wait one second

	for i in 0:10
		x = -.5 + (i*0.1)
		setPos(myCirc, [x, 0])
		draw(myCirc)                           # everything draws into memory
		#println(x)
		flip(win)                              # copies to screen

		waitTime(win, 0.1)							# 0.10

	end

	closeWinOnly(win)

	#core.quit()
	#sys.exit(0)
	#exit()

end
#-===============================================================


main()

